hw/ip/kmac/pre_syn/syn_yosys.sh

```diff
@@ -0,0 +1,182 @@
+    # where available.
+    sed -i 's/prim_flop_2sync/prim_generic_flop_2sync/g' $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i 's/prim_flop/prim_xilinx_flop/g'              $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i 's/prim_sec_anchor_flop/prim_xilinx_flop/g'   $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i 's/prim_buf/prim_xilinx_buf/g'                $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i 's/prim_sec_anchor_buf/prim_xilinx_buf/g'     $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i 's/prim_xor2/prim_xilinx_xor2/g'              $LR_SYNTH_OUT_DIR/generated/${module}.v
+done
+
+# Rename the prim_sparse_fsm_flop module. For some reason, sv2v decides to append a suffix.
+sed -i 's/module prim_sparse_fsm_flop_.*/module prim_sparse_fsm_flop \(/g' \
+    $LR_SYNTH_OUT_DIR/generated/prim_sparse_fsm_flop.v
+
+# Get and convert core sources.
+for file in "$LR_SYNTH_SRC_DIR"/rtl/*.sv; do
+    module=`basename -s .sv $file`
+
+    # Skip packages
+    if echo "$module" | grep -q '_pkg$'; then
+        continue
+    fi
+
+    # Skipe certain problematic module files that we don't need for the keccak_2share module.
+    if [ "$module" = "kmac_app" ]; then
+        continue
+    fi
+
+    sv2v \
+        --define=SYNTHESIS \
+        "${OT_DEP_PACKAGES[@]}" \
+        "$LR_SYNTH_SRC_DIR"/rtl/*_pkg.sv \
+        -I"$LR_SYNTH_SRC_DIR"/../prim/rtl \
+        $file \
+        > $LR_SYNTH_OUT_DIR/generated/${module}.v
+
+    # Make sure auto-generated primitives are resolved to generic or Xilinx-specific primitives
+    # where available.
+    sed -i 's/prim_flop_2sync/prim_generic_flop_2sync/g' $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i 's/prim_flop/prim_xilinx_flop/g'              $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i 's/prim_sec_anchor_flop/prim_xilinx_flop/g'   $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i 's/prim_buf/prim_xilinx_buf/g'                $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i 's/prim_sec_anchor_buf/prim_xilinx_buf/g'     $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i 's/prim_xor2/prim_xilinx_xor2/g'              $LR_SYNTH_OUT_DIR/generated/${module}.v
+
+    # Rename prim_sparse_fsm_flop instances. For some reason, sv2v decides to append a suffix.
+    sed -i 's/prim_sparse_fsm_flop_.*/prim_sparse_fsm_flop \#(/g' \
+        $LR_SYNTH_OUT_DIR/generated/${module}.v
+
+    # Remove the StateEnumT parameter from prim_sparse_fsm_flop instances. Yosys doesn't seem to
+    # support this.
+    sed -i '/\.StateEnumT(logic \[.*/d' $LR_SYNTH_OUT_DIR/generated/${module}.v
+    sed -i '/\.StateEnumT_StateWidth(.*/d' $LR_SYNTH_OUT_DIR/generated/${module}.v
+done
+
+#-------------------------------------------------------------------------
+# run Yosys synthesis
+#-------------------------------------------------------------------------
+yosys -c ./tcl/yosys_run_synth.tcl |& teelog syn || {
```

Comment:
This segfaults if the `nangate.lib` is not set correctly. As the flow is very experimental anyways, this might be ok.